### PR TITLE
Added pre-commit use cases in readme

### LINKED ISSUE #864: Add section to README about setting up pre-commit

### PR DESCRIPTION
**Pull request by dudiiiiiiii**
2 use cases of pre-commit added to readme. I also thought about moving them to 2 separate tabs as it was suggested on discord regarding windows and linux commands.

### CODE REVIEW

**Comment by grzanka** [comment]
The style looks like Chat-GPT, can you be more compact ? 

For us main usage of pre-commit is to run yapf. Flake8 is not used. Inspect pre-commit config and check readme.
See https://github.com/yaptide/yaptide/blob/master/.pre-commit-config.yaml
There is also check for empty files, to protect us from commiting `.env` with secrets - try that yourselves and describe in readme.

Put more emphasis on reasons why we use something. Describe how to run pre-commit hooks manually. It is also an useful way.

File: README.md [hunk 1]
```diff
@@ -208,6 +208,42 @@ To avoid running them manually we strongly recommend to use pre-commit hooks. To
 poetry run pre-commit install
 ```
 
+### Pre-commit Use Cases
+
+### Case 1: All Hooks Pass Successfully
+
+1. **Commit Changes**: Commit your changes using `git commit`.
```
Comment (grzanka) [inline README.md]: What about ide? Vscode

**Comment by grzanka** [comment]
Screenshot 2024-03-11 at 19 04 20

Do we use flake8 ?

**Comment by dudiiiiiiii** [comment]
Flake8 is configured in poetry but it isn't used so i supose i can delete this part from readme.

**Comment by grzanka** [comment]
> Flake8 is configured in poetry but it isn't used so i supose i can delete this part from readme.

You can remove config from pyproject.toml as well.

**Comment by dudiiiiiiii** [comment]
I've seen that flake8 also shows up in setup.cfg file from converter repo and is not used in pre-commit there also. Should i create issue for it?

**Comment by grzanka** [comment]
> I've seen that flake8 also shows up in setup.cfg file from converter repo and is not used in pre-commit there also. Should i create issue for it?

Good idea

**Review by grzanka — CHANGES_REQUESTED** [review]

**Review by grzanka — APPROVED** [review]

**Pull request merged**